waiter/src/waiter/scheduler/shell.clj

```diff
@@ -759,12 +759,13 @@
   "Detects and marks lost processes."
   [running-pids id->instance]
   (pc/map-vals
-    (fn [{:strs [killed? shell-scheduler/pid] :as instance}]
-      (cond-> (-> (pc/map-keys keyword instance)
-                  (update :started-at str-to-date-safe))
-        (and (not killed?) (not (contains? running-pids pid)))
-        (assoc :killed? true
-               :message "Process lost after restart")))
+    (fn [{:strs [killed?] :as instance}]
+      (let [pid (get instance "shell-scheduler/pid")]
```

Comment:
if you move this back to the destructuring above, then we don't need this `let`.

Reply:
removed.